gensim/summarization/mz_entropy.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+
+from gensim.summarization.textcleaner import tokenize_by_word as _tokenize_by_word
+from gensim.utils import to_unicode
+import numpy
+import scipy
+
+def mz_keywords(text,blocksize=1024,scores=False,split=False,weighted=True,threshold=0.0):
+    """Extract keywords from text using the Montemurro and Zanette entropy algorithm.
+       https://arxiv.org/abs/0907.1558
+       :param text: str (document to summarize)
+       :param blocksize: int (size of blocks to use in analysis)
+       :params scores: bool (return score with keywords)
+       :params split: bool (return results as list)
+       :params weighted: bool (weight scores by word frequency. 
+                               False is useful for shorter texts)
+       :params threshold: float or 'auto' (minimum score for returned keywords
+                                           'auto' calculates the threshold as 
+                                           nblocks/(nblocks+1.0)
+                                           Use 'auto' with weighted=False)"""
+    text=to_unicode(text)
```

Comment:
Many PEP8 issues (in each line), look to [checker log](https://travis-ci.org/RaRe-Technologies/gensim/jobs/306587389#L518).